[FIX] hr_holidays_public: Consider public holdays for leaves in hours

maisim opened the pull request:
Backport of

7d5bc92 [FIX] hr_holidays_public: Exclude public holidays on hours computation on validated leave
887fd4a6 [FIX] hr_holidays_public: Being multi-record aware in _compute_number_of_hours_display

from 13.0 ( hr-holidays repo )

Make it working on 12.0

Refactor tests files: 

- Move to TransactionCase
- Split  test file :
  - one common file (the common setUp) 
  - one file for tests on holidays in days
  - one file in holidays in hours

santostelmo committed:
[FIX] hr_holidays_public: Exclude public holidays on hours computation on validated leave

backport 7d5bc92 from hr-holidays repo

pedrobaeza committed:
[FIX] hr_holidays_public: Being multi-record aware in _compute_number_of_hours_display

backport of 887fd4a6 from hr-holidays repo

pedrobaeza commented on hr_holidays_public/__manifest__.py:
```diff
@@ -2,26 +2,24 @@
-        'views/hr_holidays_public_view.xml',
-        'views/hr_leave_type.xml',
-        'wizards/holidays_public_next_year_wizard.xml',
+    "name": "HR Holidays Public",
```
Please reduce the diff to the minimal needed.

maisim commented on hr_holidays_public/models/hr_leave.py:
```diff
@@ -25,10 +25,18 @@ def _get_number_of_days(self, date_from, date_to, employee_id):
+        For such cases, we need to serialize the call to super in fragments.
+        """
+        to_serialize = self.filtered(
+            lambda x: x.state == "validate"
```
@pedrobaeza 

Did you remember why you filter on state ? It make the displayed value in the form false (when we create a new leave)

You can check here (today is public holiday ! :p )
http://oca-hr-holidays-13-0-d4d9fdf82183.runboat.odoo-community.org/web#action=168&model=hr.leave&view_type=list&cids=&menu_id=117

When I remove this filter it's ok but I cannot undestand what it is for

pedrobaeza replied:
The explanation is in the method docstring.

maisim replied:
I saw, but has I understood how it work we always need to push exclude_public_holidays=True in the context to get the good number of hours whatever the state

pedrobaeza replied:
Not on v13 at least, and not if the holidays type doesn't have the check.

maisim replied:
Today is defined as public holiday
![image](https://user-images.githubusercontent.com/6648433/155359895-cdaa10e4-0dbe-4b9d-b028-9ecb3a43af23.png)

"Compensatory days" type exclude public holidays
![image](https://user-images.githubusercontent.com/6648433/155360640-4d7ec1c6-a217-4d46-aa40-c8d44faaed8e.png)

This is not taken into account in the form view
![image](https://user-images.githubusercontent.com/6648433/155360187-f24b810e-e805-4466-a576-f0671bc22271.png)

From http://oca-hr-holidays-13-0-d4d9fdf82183.runboat.odoo-community.org/web#action=168&model=hr.leave&view_type=list&cids=&menu_id=117

I have the same behavor on 12.0. Remove state filter fix the problem

I'm really sorry if there's something I don't understand and I'm wasting your time but I feel like it's not working the way it should

pedrobaeza replied:
OK, then remove it such condition. Have you seen if in v13 is happening the same problem?

maisim replied:
Ok it can't work as is on a vanilla 12.0

Let me take a closer look, I will let you know when the PR is ready.

For 13.0, if you confirm the problem I can open an issue or submit a PR

pedrobaeza replied:
That's the question. I don't think there's an issue in v13. The problem is here on v12.

maisim replied:
The screenshots are from the link above (13.0 runboat instance)

maisim committed:
[FIX] hr_holidays_public: make Exclude public holidays on hours working
on 12.0
[IMP] hr_holidays_public: refactor tests files
[IMP] hr_holidays_public: add tests

maisim commented:
Hi @pedrobaeza ,
Can you please have a look on it ? 
I don't know what is the good way to get all green as the only solution a can see is to add tests to not concerned by this PR parts of code to improve coverage
Thanks in advance for your help

nimarosa reviewed (changes requested):
Don't change version number in manifest. The rest LGTM.

maisim commented:
> Don't change version number in manifest. The rest LGTM.

Hi @nimarosa, Thanks for your review.
Just to understand, can you explain why the version number should not change ?
This PR is a bugfix and will change the behavior of the plugin (requests in hours should work correcty), In my mind it is a case where the version number have to change.

If I am wrong, Is https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#version-numbers need an update?

mtelahun commented:
The version numbers will be automatically bumped by ocabot 
(https://github.com/OCA/oca-github-bot) when your PR is merged.

On 16/09/2022 09:31, Simon Maillard wrote:
>
>     Don't change version number in manifest. The rest LGTM.
>
> Hi @nimarosa <https://github.com/nimarosa>, Thanks for your review.
> Just to understand, can you explain why the version number should not 
> change ?
> This PR is a bugfix and will change the behavior of the plugin 
> (requests in hours should work correcty), In my mind it is a case 
> where the version number have to change.
>
> If I am wrong, Is 
> https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#version-numbers 
> need an update?
>
> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/OCA/hr/pull/1079#issuecomment-1248970394>, or 
> unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AB4YOHG7B2XEYGMRRIOTRQTV6QH4JANCNFSM5PEKDT6Q>.
> You are receiving this because you are subscribed to this 
> thread.Message ID: ***@***.***>
>

nimarosa commented:
/ocabot merge patch

nimarosa commented:
/ocabot merge patch

maisim commented:
ping @OCA/human-resources-maintainers

Any chance to get it merged ? Let me know if I can do something to help

SirTakobi commented:
@OCA/human-resources-maintainers ^

nimarosa commented:
Let's try again. Codecov should not block the merge. 

/ocabot merge patch
